docs/key_management.md

```diff
@@ -0,0 +1,176 @@
+                //seed has been created and is accessible using rootAddress
+                // * the handle is `rootAddress`
+                // * the corresponding publicKey in base64 is `publicKey`
+                // * if there was an error, those are blank and the err object is non null
+
+                // To use the seed, refer to it using this `rootAddress`
+            })
+```
+
+You can also import bip39 mnemonic phrases:
+
+```kotlin
+
+//bip39 mnemonic phrase:
+val phrase = "vessel ladder alter ... glass valve picture"
+
+UportHDSigner().importHDSeed(activity, KeyProtection.Level.SIMPLE, phrase, { err, rootAddress, publicKey ->
+
+                if (err != null) {
+                    //handle error
+                } else {
+                    assertEquals("0x794a...e96ac8", rootAddress)
+                    //seed has been imported and
+                    // * the handle is `rootAddress`
+                    // * the corresponding publicKey in base64 is `publicKey`
+                }
+
+            })
+```
+
+#### Signing
+
+You can use this lib to calculate ETH transaction signatures.
+Building and encoding transaction objects into `ByteArray`s is not in the scope of this lib.
+
+You can sign transactions using keys derived from a previously imported seed.
+To refer to that seed you must use the `rootAddress` from the seed creation/import callback
+Based on the `KeyProtection.Level` used during seed import/creation, a prompt may be shown to the user
+on the lock-screen / fingerprint dialog.
+
+```kotlin
+
+val rootAddress = "0x123..." //rootAddress received when creating/importing the seed
+
+//bip32 key derivation
+val derivationPath = "m/44'/60'/0'/0/0"
+
+//the transaction payload, base64 encoded
+val txPayloadB64 = Base64.encodeToString( transaction.rlpEncode(), Base64.DEFAULT )
+
+//gets shown to the user on fingerprint dialog or on lockscreen, based on `KeyProtection.Level` used
+val prompt = "Please sign this transaction"
+
+UportHDSigner().signTransaction(activity, rootAddress, derivationPath, txPayloadB64, prompt, { err, sigData ->
+    if (err != null) {
+        //handle error
+    } else {
+        //use sigData r,s,v components
+    }
+})
+
+```
+
+Note: The requirement to encode the transaction payload as a base64 string is subject to change in future releases
+
+##### uPort specific JWTs
+
+This lib can also produce signatures for uPort specific JWTs:
+The method signature is the same but the signing method differs.
+
+Also, do note that in the current version of this API,
+ `data` is a Base64 encoded string (not a serialized JSON) - this is subject to change in future releases
+
+```kotlin
+
+UportHDSigner().signJwtBundle(activity, rootAddress, derivationPath, data, prompt, { err, sigData ->
+    if (err != null) {
+        //handle error
+    } else {
+        //use sigData r,s,v components
+    }
+})
+
+```
+
+
+### Changelog
+
+
+#### latest
```

Comment:
@mirceanis I think we should remove this or update it with most recent changes if any

Reply:
yes, exactly.
please remove the changelog section